src/compiler/preprocess/index.ts

```diff
@@ -43,16 +43,18 @@ interface Replacement {
 async function replace_async(str: string, re: RegExp, func: (...any) => Promise<string>) {
 	const replacements: Array<Promise<Replacement>> = [];
 	str.replace(re, (...args) => {
-		replacements.push(
-			func(...args).then(
-				res => // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
-					({
-						offset: args[args.length - 2],
-						length: args[0].length,
-						replacement: res,
-					}) as Replacement
-			)
-		);
+		if (args.slice(1, args.length - 2).some(Boolean)) {
```

Comment:
I'm still not sure what the changes to `replace_async` are for. The intention was that it be a low-level utility function, behaving a lot like `String.prototype.replace`, except able to work with callbacks that return promises. If you need different behavior than that, it should be handled in the callback you pass to `replace_async`.

Reply:
oh yea you are right. i've updated my changes.